templates/server/puppetserver/conf.d/auth.conf.erb

```diff
@@ -12,7 +12,7 @@ authorization: {
                 type: regex
                 method: [get, post]
             }
-            allow: <%= @server_trusted_agents << '$1' %>
+            allow: <%= @server_trusted_agents + ['$1'] %>
```

Comment:
It's not just in this template that `server_trusted_agents` was being mutated, but in the catalog itself. So much for puppet variables being immutable! I guess in irb, you can accidentally abuse this. ;)

Reply:
oops.  That was me! https://github.com/theforeman/puppet-puppet/commit/4f07b5fac070f1c9ee8762df3852787f1d0558f3#diff-6775a3530d5d8059f4175382a1af402cR12

Reply:
Also, 'naming things is hard'.  Maybe I should have called it `trusted_compile_clients` or something instead.